waspc/data/Generator/templates/react-app/src/index.js

```diff
@@ -19,6 +22,7 @@ async function startApp() {
   {=# doesClientSetupFnExist =}
   await {= clientSetupJsFnIdentifier =}()
   {=/ doesClientSetupFnExist =}
+  setupQueryClient()
```

Comment:
We definitely need to call _something_ that initializes the `queryClient` (i.e., makes sure to resolve the `queryClientInitialized` promise) if the user doesn't initialize it themselves.

I thought about two approaches:
1. Initialize the `queryClient` on `setupQueryFunction`'s **first** invocation, ignoring all subsequent calls (this is what I currently have)
2. Write two functions:
    - `setupQueryClient` - A function that persists the config objects somewhere inside the `queryClient` module using a closure. This is the function users would call it from the client setup function.
    - `initializeQueryClient` - A function that actually calls the constructor and uses the previously persisted state (or the default one). This is the function we would always be calling from here.

I don't have a preference between those two, each has pros and cons. Feel free to leave your opinions in this thread :)

Reply:
Yup, I think that is a bit tricky -> if you allow me to call it multiple times, I would expect the multiple calls to all be effective, not that subsequent calls will be ignored.
So to avoid possibly surprising behaviour, I would either throw an error on subsequent calls, or actually re-setup the query client.
The question is, what do we do with the fact that they might not call the setup?
There is also the fact now that getting query client now is async, so we had to make many functions async that use it.

How about this:
1. In queryClient.js, we by default set `queryClient` to default query client. So same like we had so far. We export `queryClient`, it is const, and it is not a Promise, it is normal value.
2. We expose the async function `setupQueryClient` to the Wasp users. This functions resolves once it is done with setting up the queryClient. It can be called multiple times, and it will set the query client again each time. They can call it in the client.setup function if they want, or do whatever they want with it.

Shouldn't this be an ok solution?
This way behaviour is pretty simple + we don't have to await `queryClient` through the whole code when we already know it is available.
And if they want to make sure that the whole app uses their own queryClient, they just need to await their call of `setupQueryClient` in the `client.setup` and they are good.

Reply:
Sounds good, that should address the multiple call problem.